test/e2e/conftest.py

```diff
@@ -220,6 +220,51 @@ def session_worker(
     stop_worker(request, worker)
 
 
+@pytest.fixture(scope="class", params=[True, False])
+def asset_sync_worker_config(
+    request: pytest.FixtureRequest,
+    posix_job_user: PosixSessionUser,
+    posix_env_override_job_user: PosixSessionUser,
+    posix_config_override_job_user: PosixSessionUser,
+    worker_config: DeadlineWorkerConfiguration,
+    windows_job_users: list[str],
+) -> DeadlineWorkerConfiguration:
+    """
+    Parametrized worker configuration fixture that tests ASSET_SYNC_JOB_USER_FEATURE with both True and False values.
+
+    This fixture ensures the environment variable is set before worker initialization,
+    allowing proper testing of the asset sync feature flag behavior.
+    """
+    asset_sync_feature = request.param
+    worker_env_var = {
+        "ASSET_SYNC_JOB_USER_FEATURE": str(asset_sync_feature),
+    }
+    LOG.info(f"worker_env_var: {worker_env_var}")
+
+    return dataclasses.replace(
+        worker_config,
+        job_users=[
+            posix_job_user,
+            posix_config_override_job_user,
+            posix_env_override_job_user,
+        ],
+        windows_job_users=windows_job_users,
+        worker_env_var=worker_env_var,
```

Comment:
Maybe we should extend the existing env var instead of overriding the entire dict.
```suggestion
        worker_env_var={
            **worker_config.worker_env_var,
            "ASSET_SYNC_JOB_USER_FEATURE": str(asset_sync_feature)
        },
```

Reply:
Updated above.